File: mo-system/mo-system-api/src/main/java/ink/ckx/mo/system/api/model/vo/dept/DeptListVO.java
package ink.ckx.mo.system.api.model.vo.dept;

import com.fasterxml.jackson.annotation.JsonFormat;
import ink.ckx.mo.common.web.enums.StatusEnum;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Schema(description = "部门列表VO")
@Data
public class DeptListVO {

    @Schema(description = "部门id")
    private Long id;

    @Schema(description = "父部门id")
    private Long parentId;

    @Schema(description = "部门名称")
    private String name;

    @Schema(description = "排序")
    private Integer sort;

    @Schema(description = "状态")
    private StatusEnum status;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @Schema(description = "更新时间")
    private LocalDateTime updateTime;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @Schema(description = "创建时间")
    private LocalDateTime createTime;

    @Schema(description = "子部门")
    private List<DeptListVO> children;
}

File: mo-admin/mo-admin-api/src/main/kotlin/ink/ckx/mo/admin/api/model/vo/dept/DeptListVO.kt
package ink.ckx.mo.admin.api.model.vo.dept

import ink.ckx.mo.common.web.enums.StatusEnum
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

@Schema(description = "部门列表VO")
data class DeptListVO(

    @Schema(description = "部门id")
    var id: Long? = null,

    @Schema(description = "父部门id")
    var parentId: Long? = null,

    @Schema(description = "部门名称")
    var name: String? = null,

    @Schema(description = "排序")
    var sort: Int? = null,

    @Schema(description = "状态")
    var status: StatusEnum? = null,

    @Schema(description = "更新时间")
    var updateTime: LocalDateTime? = null,

    @Schema(description = "创建时间")
    var createTime: LocalDateTime? = null,

    @Schema(description = "子部门")
    var children: List<DeptListVO>? = null,
)